src/main/java/org/commcare/formplayer/application/MenuController.java

```diff
@@ -96,7 +96,8 @@ public EntityDetailListResponse getDetails(@RequestBody SessionNavigationBean se
                     sessionNavigationBean.getQueryDictionary(),
                     sessionNavigationBean.getOffset(),
                     sessionNavigationBean.getSearchText(),
-                    sessionNavigationBean.getSortIndex()
+                    sessionNavigationBean.getSortIndex(),
+                    false
```

Comment:
All three of the `advanceSessionWithSelections` calls need to use `sessionNavigationBean.isDoQuery()`, otherwise the case search request itself will do the query, but the subsequent `get_details` request won't and so you can't select the case and end up [here](https://github.com/dimagi/formplayer/blob/64ec42d0a7540c0c2e6f997a34bc25924a1af6b4/src/main/java/org/commcare/formplayer/application/MenuController.java#L132). (It's crazy that the search request gets sent multiple times, but that's a separate issue.)

For what it's worth: I ran into this because I checked this branch out to see how it would interact with a bug I'm fixing, which is that case search doesn't work properly in workflows where you select a parent case and then a child case. The root problem with that bug is that formplayer only has one `queryDictionary`, so it can't handle multiple searches. My approach is turning `queryDictionary` into a map where keys are based on `selections` (like `2,action 1`) and the values are hashtables like the current `queryDictionary`. It's a bit convoluted. Similarly, I expect `doQuery` may have multiple values: you may want to do the query for parent selection but not for child selection. So I'm expecting to turn `doQuery` into a map, again keyed by `selections`.

Reply:
Ahh I didn't realize there is another subsequent search call. Thanks for highlighting. 

> case search doesn't work properly in workflows where you select a parent case and then a child case.

Ahh interesting. Your solution sounds right to me. I would be happy to make those changes if needed.

Reply:
Thank you, I think I'm almost there, should have a PR up shortly 🙂